bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/servlet/IpCameraServlet.java

```diff
@@ -59,6 +60,8 @@ public IpCameraServlet(ThingHandler handler, HttpService httpService, Dictionary
 
     public void startListening() {
         try {
+            Hashtable<Object, Object> initParameters = new Hashtable<>();
```

Comment:
You are completely overriding the `initParameters` provided by the second constructor in lines 54-59. This constructor is used from this subclass:

https://github.com/openhab/openhab-addons/blob/275329d4858acf5067c74bf6a1a5d88b91f88f93/bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/servlet/CameraServlet.java#L47-L56

so it will break `async-supported` being set to `true`.

You could instead clone `this.initParameters` and add the additional entry.

Reply:
Perhaps my java terminology is poor, but you don't need to clone it, you can just delete the new import and the following line to directly use initParameters.

remove this line...
```
Hashtable<Object, Object> initParameters = new Hashtable<>();
```

You can use the initParameters.put directly leaving it as a single line change. I have tested this tonight and it fixes the issue here. Thanks for the PR.

Reply:
Added commit to address comment

Reply:
@Skinah - you are right, thanks for the proposal which I see is adopted by @apella12.

In general I'm hesitant to manipulate collections provided by external callers in case they would not expect this when not clearly documented, for example by a method signature like `void addAdditionalParameters(Dictionary<Object, Object> parameters)`.